### PR TITLE
Add browserTZ option which allows the timezone to be the local browser timezone instead of UTC. 

### PR DESCRIPTION
**Pull request by AJRepo**
Thanks for the library. Worked almost perfectly except that I was asked to implement it for a situation in which the timezone wasn't specified in the date/time of the event. 

E.g. the data to be presented only had date, time and said "your local timezone." 

Given the nature of the setup - changing the business logic to force timezones into the existing legacy display of data wasn't an option. 

This feature request  creates a new option "browserTz" which can override the specified timezone. If used it will create a link that uses the local users' timezone. Basically it strips the timezone field out of the ical/url link which causes the event to default to the browser/user's timezone. If that option is not specified then it does not change existing functionality. 

## Types of changes

-   [ ] Bug fix (non-breaking change which fixes an issue)
-   [X ] New feature (non-breaking change which adds functionality)
-   [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

-   [X ] My code follows the code style of this project.
-   [X] My change requires a change to the Readme.md.
-   [X] I have updated the Readme.md accordingly.
-   [X] I have read the **CONTRIBUTING** document.

### CODE REVIEW

**Comment by jekuer** [comment]
@AJRepo, thanks for the PR.
Provocative question: At which use case does it make sense to use the browser's time zone? This would mean that the time of an event can differ per user. So, for user 1 it could start at 5pm, while for user 2 the same event starts at 8pm.

**Comment by AJRepo** [comment]
The specific use case I was solving was a website which lists local events across multiple time zones with in-person attendance. People sign up for the event online but attend in person.  Events can be scheduled 6-12 months in advance, so if one is picking a timezone now as UTC+N that can become UTC+N+/-1  after a daylight savings time shift.  Some places have exemptions from daylight savings which can add complexity to a person guessing the correct timezone.  

This means it falls on the person adding the events to figure out the full timezone for each event listed when they are not perhaps familiar with that location. That adds work cycles, potential errors, and complexity. It was easier with fewer mistakes made to just specify a time like "10:00 am at your local timezone."   The side effect of removing friction/complexity from data entry is that there is now a database of times where the timezone field is not used on purpose.

One can imagine other uses cases. E.g.

* A local calendar/alarm on a laptop which does not reset timezones while traveling. 

* A calendar/alarm for a traveling person where you don't know what timezone they are in and yet want to set the time based on the local timezone (e.g. take pills at 10 am)

**Comment by jekuer** [comment]
For your example, you can already use the "timeZone" options. It automatically respects Daylight savings.

I am not sure if I am missing something, but I do not see the problem:
You have an event, which starts at 9am PST. So, you simply set the timeZone to Los Angeles (for example). No matter where on the world I save this event to my calender, it will have the correct time, since the PST-Time gets converted to UTC and my calender simply saves this normalized value.
On the other side, if I take the browser time, I would alter the base of the calculation, which would lead to people having the wrong (!) time in their calendars.
Mind that the timeZone option defines the timezone of the event's location, not of the user (this would be indeed crazy)!

Bottom line: No matter if I am now mixing things up, the timezone should be statically defined by the JSON data, NEVER dynamically by the browser.

What could be improved is the iCal generation. There, it would make sense to use the existing timezone features (see #1 ).

**Comment by AJRepo** [comment]
> I am not sure if I am missing something, but I do not see the problem:

Two main problems

1) Forcing a timezone puts the burden of knowledge of the timezone for each tiny town scattered across the surface of the earth on the person entering events. Let's say you are told there will be an event in Tuba City, AZ .  What timezone do you pick? 

If you have to look that up, you start to see the issue when you multiply that by tons of events for a person who's not familiar with those areas. Stating "In your local time zone" allows the person doing data entry to not have to lookup which timezone to use for Tuba City, AZ or every tiny town everywhere. 

2) Without this upgrade, this library does not match the functionality for data entry in calendaring systems (e.g. Google Calendar) that support a browser-defined timezone.

**Comment by jekuer** [comment]
Now I got it. We are talking of 2 different use case.

The one, the script was originally intended for:
The person that builds the website has an event he/she wants to share. Maybe worldwide.

- Timezone is well known.
- It is also important to set, because otherwise, people worldwide would receive different times, which would be wrong.

This should be the usual (95%) use case for such a library.


**Your case:**
There is no static event, but rather some kind of reminder, which is individual to the respective user AND his CURRENT time zone (otherwise, if he would move to the next time zone, the old event in his Google Cal would be wrong again).
Like your "take the pill at 10am" example (get it now).

> I would prefer a more simpler solution (especially since it's an edge case). Will follow up on this in a few minutes.

**Comment by jekuer** [comment]
@AJRepo , please check the following branch for a smaller solution of the issue:
https://github.com/jekuer/add-to-calendar-button/tree/timezone_tweak

Feel free to comment if this does not cover your case. I will close this PR here and merge the mentioned branch as soon as you comment or latest in the next days.

Thanks for your input.
